examples/examples/encode-microphone/encode-microphone.rs

```diff
@@ -0,0 +1,343 @@
+            ..Default::default()
+        }],
+        ..Default::default()
+    };
+
+    // Create a new RTCPeerConnection
+    let peer_connection = Arc::new(api.new_peer_connection(config).await?);
+
+    let notify_tx = Arc::new(Notify::new());
+    let notify_audio = notify_tx.clone();
+
+    let (done_tx, mut done_rx) = tokio::sync::mpsc::channel::<()>(1);
+    let audio_done_tx = done_tx.clone();
+
+    // Create a audio track
+    let audio_track = Arc::new(TrackLocalStaticSample::new(
+        RTCRtpCodecCapability {
+            mime_type: MIME_TYPE_OPUS.to_owned(),
+            ..Default::default()
+        },
+        "audio".to_owned(),
+        "webrtc-rs".to_owned(),
+    ));
+
+    // Add this newly created track to the PeerConnection
+    let rtp_sender = peer_connection
+        .add_track(Arc::clone(&audio_track) as Arc<dyn TrackLocal + Send + Sync>)
+        .await?;
+
+    // Read incoming RTCP packets
+    // Before these packets are returned they are processed by interceptors. For things
+    // like NACK this needs to be called.
+    tokio::spawn(async move {
+        let mut rtcp_buf = vec![0u8; 1500];
+        while let Ok((_, _)) = rtp_sender.read(&mut rtcp_buf).await {}
+        Result::<()>::Ok(())
+    });
+
+    let (sender, frame_receiver) = flume::bounded::<AudioFrame>(3);
+    let (encoded_sender, encoded_receiver) = flume::bounded::<AudioEncodedFrame>(3);
+
+    // Encoder thread
+    thread::spawn(move || {
+        // We just handle 48khz, to handle other sample rates like 44.1khz you need to use a resampler.
+        let mut encoder =
+            opus::Encoder::new(48000, opus::Channels::Mono, opus::Application::Voip).unwrap();
+
+        loop {
+            let AudioFrame { data } = frame_receiver.recv().unwrap();
+
+            let sample_count = data.len() as u64;
+            // sample duration
+            let duration = Duration::from_millis(sample_count * 1000 / 48000);
+            let encoded = encoder
+                .encode_vec_float(&data, 1024)
+                .expect("Failed to encode");
+            let bytes = Bytes::from(encoded);
+
+            encoded_sender
+                .send(AudioEncodedFrame { bytes, duration })
+                .unwrap();
+        }
+    });
+
+    // STREAM
+    let device = get_default_input_device().expect("Failed to get default device.");
+
+    // ---
+    let input_configs = match device.supported_input_configs() {
+        Ok(f) => f,
+        Err(e) => {
+            panic!("Error getting supported input configs: {:?}", e);
+        }
+    };
+    let input_configs2 = input_configs
+        .into_iter()
+        .find(|c| c.max_sample_rate() == SampleRate(48000))
+        .expect("did not find a sample rate of 48khz");
+
+    let config = input_configs2.with_sample_rate(SampleRate(48000));
+
+    let err_fn = move |err| {
+        eprintln!("an error occurred on stream: {}", err);
+    };
+
+    // until it is 960
```

Comment:
make a comment what 960 is, my guess it's 20ms frame with 48kHz sampled data.

Reply:
Correct, I should replace it with a const and a comment.